File: api/src/main/java/com/ricky/adocao/models/Usuario.java
package com.ricky.adocao.models;

import com.ricky.adocao.enums.RolesEnum;
import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
public class Usuario {
    @Id
    @Column(name = "USER_ID")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "LOGIN", length = 50)
    private String login;

    @Column(name = "SENHA", length = 50)
    private String senha;

    @Column(name = "ROLE", length = 10)
    @Enumerated(EnumType.STRING)
    private RolesEnum role;

    @Column(name = "TELEFONE", length = 10)
    private String telefone;

    @Column(name = "EMAIL", length = 50)
    private String email;
}

File: src/main/kotlin/com/ricky/adocao/models/Usuario.kt
package com.ricky.adocao.models

import com.ricky.adocao.enums.RoleEnum
import jakarta.persistence.*
import lombok.Builder
import lombok.NoArgsConstructor
import org.springframework.security.core.GrantedAuthority

@Entity(name = "USUARIO")
data class Usuario(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "USER_ID")
    var id: String = "",

    @Column(name = "NOME", length = 50)
    var nome: String = "",

    @Column(name = "LOGIN", length = 50)
    var login: String = "",

    @Column(name = "SENHA")
    var senha: String = "",

    @Column(name = "EMAIL", length = 50)
    var email: String = "",

    @Column(name = "TELEFONE", length = 10)
    var telefone: String = "",

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
        name = "USUARIO_ROLE",
        joinColumns = [JoinColumn(name = "USER_ID")],
        inverseJoinColumns = [JoinColumn(name = "ROLE_ID")]
    )
    var roles: List<Role> = mutableListOf()

)
